### PR TITLE
M Mykytovych Java8

### PR DESCRIPTION
**Pull request by marjana07**

### CODE REVIEW

File: src/main/java/parking/VehicleParking.java [hunk 1]
```diff
@@ -12,30 +16,59 @@
...
-    public static boolean isNumberExist(String template) {
-        for (AbstractVehicle vehicle : vehicleList) {
-            if (vehicle.getNumber().substring(0, 2).equals(template)) {
-                return true;
-            }
+    /* Task #4   Check if at least one vehicle in your list corresponds to some
+     search criteria (e.g. vehicle is registered in Lviv (plate number starts with ‘BC’))
+     */
+    public static boolean isNumberExist(String number) {
+        return vehicleList.stream().filter(o -> o.getNumber().substring(0, 2).equals(number)).findFirst().isPresent();
```
Comment (olobanok) [inline src/main/java/parking/VehicleParking.java]: you may replace findFirst().isPresent() with anyMatch()

File: src/main/java/parking/VehicleParking.java [hunk 2]
```diff
@@ -12,30 +16,59 @@
...
+    public static boolean isAllTypes(String typeSearch) {
+        return (vehicleList.stream().noneMatch(o -> o.getVehicleType().equals(typeSearch)));
+    }
+
+    /* Task #7  Traverse vehicle list, collect all unique registration numbers,
+    sort them alphabetically and print them to console using comma as a delimiter
+     */
+    public static String getCarByUniqueNumber() {
+
+        sortedNumbersList = vehicleList.stream().map(vehicleList -> vehicleList.getNumber()).distinct().sorted().collect(Collectors.toList());
```
Comment (olobanok) [inline src/main/java/parking/VehicleParking.java]: lambda vehicleList -> vehicleList.getNumber() can be replaced with method reference

Comment (marjana07) [reply src/main/java/parking/VehicleParking.java]: done

File: src/main/java/parking/VehicleParking.java (hunk 1 above)
Comment (marjana07) [reply src/main/java/parking/VehicleParking.java]: done

**Pull request merged**